integration_test/Bindings/Python/dialects/hw.py

```diff
@@ -48,8 +48,8 @@ def build(module):
       # CHECK: [[ARRAY2:%.+]] = hw.array_create %[[CONST]] : i32
       array2 = hw.ArrayCreateOp.create([constI32])
 
-      # CHECK: %false = hw.constant false
-      # CHECK: hw.array_get [[ARRAY2]][%false] : !hw.array<1xi32>
+      # CHECK: %c0_i0 = hw.constant 0 : i0
+      # CHECK: hw.array_get [[ARRAY2]][%c0_i0] : !hw.array<1xi32>
```

Comment:
This needs index type.

Reply:
I deliberately did not add index types to checks on `hw.array_get` ops. My hope is that we'll remove this special, ambiguous type case as soon as possible, and when that is done, it would be nice to not have to go and refactor all these things again...